halyard-config/src/main/java/com/netflix/spinnaker/halyard/config/model/v1/providers/aws/AwsProvider.java

```diff
@@ -20,8 +20,32 @@
 import com.netflix.spinnaker.halyard.config.model.v1.node.Provider;
 import com.netflix.spinnaker.halyard.config.model.v1.node.Validator;
 import com.netflix.spinnaker.halyard.config.problem.v1.ConfigProblemSetBuilder;
+import lombok.Data;
 
+import java.util.Collections;
+import java.util.List;
+
+@Data
 public class AwsProvider extends Provider<AwsAccount> {
+  private String accessKeyId;
+  private String secretAccessKey;
+
+  private String defaultAssumeRole;
+  private String defaultKeyPairTemplate = "{{name}}-keypair";
+
+  private List<AwsRegion> defaultRegions = Collections.singletonList(new AwsRegion().setName("us-west-2"));
```

Comment:
In a sense this is the externalized config - this only comes into play the first time your halconfig is generated after installing, and this is the only place this constant is referenced.